### PR TITLE
Make space for foorm library editor routes

### PR DESCRIPTION
**Pull request by bencodeorg**
Folder structure/routes for the current foorm editor is built to allow edits to forms, but not to library questions. This PR makes room for library question editing routes.

Three sets of routes used in current form editor (I think):
1. /api/v1/pd/foorm/form/ (used primarily to get information about a form or set of forms) -> put into their own form namespace and made plural to align with Rails standard practices, will set up parallel library namespace
2. /foorm/forms/ (used to save changes to forms) -> already well set up for library addition, nothing to change here.
3. /foorm/editor (used to initialize edit page) -> renamed controller to foorm/forms/editor, put new controller at foorms/library_questions/editor.

## Testing story

I tested that /foorm/forms/editor renders the existing page, and /foorm/library_questions/editor renders the list of options to choose from (placeholder).

I also tested that the CSV export of responses to a Foorm form on the workshop dashboard still works.

## Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/config/routes.rb
Comment (bencodeorg) [inline dashboard/config/routes.rb:513]: This will catch literally any URL that begins with /api/v1/pd/foorm/form/...?

File: dashboard/config/routes.rb
Comment (bencodeorg) [inline dashboard/config/routes.rb]: Not sure on Rails/JS singular/plural choices, if anyone has opinions.

File: dashboard/config/routes.rb
Comment (jamescodeorg) [inline dashboard/config/routes.rb]: It might be nice to keep all the foorm routes together.  Any idea why these were separated before?

Comment (jamescodeorg) [inline dashboard/config/routes.rb]: Is this route just here for backwards compatibility?  If so, might be good to add a comment?

**Review by jamescodeorg — COMMENTED** [review]
Is there a spec or something I can read to learn more about this feature?  (I'm not very familiar with foorm or libraries.)

File: dashboard/app/controllers/foorm_form_editor_controller.rb
Comment (molly-moen) [inline dashboard/app/controllers/foorm_form_editor_controller.rb]: could we do instead Foorm::FormEditorController (similarly Foorm::LibraryEditorController)? That would follow the patter of Foorm::Form and Foorm::LibraryQuestion and be more readable

File: dashboard/app/controllers/foorm_form_editor_controller.rb
Comment (molly-moen) [inline dashboard/app/controllers/foorm_form_editor_controller.rb]: nit: `foorm/form/editor` makes more sense to me, since it is the foorm::form editor

File: dashboard/app/controllers/foorm_library_editor_controller.rb
Comment (molly-moen) [inline dashboard/app/controllers/foorm_library_editor_controller.rb]: Note: this doesn't match what is in the routes. For consistency should this be `foorm/library/editor`? Or the form one could be `foorm/forms_editor`

File: dashboard/config/routes.rb
Comment (molly-moen) [reply dashboard/config/routes.rb]: Personally I like `foorm/form/editor` since we are only ever editing one form at a time and it makes more sense grammatically. But am not sure if that's the right convention.

**Comment by bencodeorg** [comment]
> Is there a spec or something I can read to learn more about this feature? (I'm not very familiar with foorm or libraries.)

Molly wrote a good summary of what the current version of the foorm editor is (basically a way to edit surveys via JSON config), happy to chat through how to get set up:
https://github.com/code-dot-org/code-dot-org/wiki/%5BLevelbuilder%5d-The-Foorm-Editor

Spec does not really exist -- short version is "create something very similar to the existing foorm editor that allows survey writers to edit library questions". The current editor allows a survey writer to edit a form, but forms can be constructed from "library questions" that can be reused across multiple surveys. These library questions have initial config stored [here](https://github.com/code-dot-org/code-dot-org/tree/staging/dashboard/config/foorm/library/surveys/pd), loaded into our DB [here](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/foorm/library_question.rb#L24), and can't be edited at the moment (except via PR).

File: dashboard/config/routes.rb
Comment (bencodeorg) [reply dashboard/config/routes.rb]: Yep, I can add a comment. I think the more common case will be wanting to edit a form vs. editing a library, so wanted to keep it as the default.

File: dashboard/app/controllers/foorm_form_editor_controller.rb
Comment (bencodeorg) [reply dashboard/app/controllers/foorm_form_editor_controller.rb]: Makes sense to me!

File: dashboard/config/routes.rb
Comment (bencodeorg) [reply dashboard/config/routes.rb]: mmm not 100% sure, the editor does feel like it's kind of separate from the other places we use foorm (there's sort of the "render a form for someone to fill out" part and "collect and do something with submissions to a given form" part). I agree, but there's kind of foorm stuff sprinkled throughout routes.rb and didn't want to bite off more cleanup than I could chew :).

**Comment by bencodeorg** [comment]
@molly-moen @jamescodeorg could you take another pass through this? Looking at your comments and doing more digging into Rails routes conventions, I made everything either `forms` or `library_questions`, as it seems like plural of model names in controllers is the standard.

**Review by molly-moen — APPROVED** [review]
Looks good! Would be good to add (either in this PR or as a follow up) a link to the library editor from the levelbuilder panel on the homepage like [this](https://github.com/code-dot-org/code-dot-org/pull/37541/files#diff-db4207c3f1dd76ce2258213247059de3bd7e2a10cbd723753640e624537ade0a).

**Review by jamescodeorg — APPROVED** [review]
LGTM (but I would wait for Molly to give a more authoritative approval :-)

**Pull request merged**
